src/model/src/ModelTransformers.cpp

```diff
@@ -805,4 +806,70 @@ bool extractSubModel(const iDynTree::Model& fullModel, const iDynTree::Traversal
     return  true;
 }
 
+void addValidNamesToAllSolidShapesHelper(iDynTree::Model& outputModel,
+                                         std::vector<std::vector<SolidShape *>> solidShapes,
+                                         std::string suffix)
+{
+    for (iDynTree::LinkIndex lnkIndex = 0; lnkIndex < outputModel.getNrOfLinks(); lnkIndex++)
+    {
+        // Check the number of solid shapes for this link
+        size_t nrOfSolidShapes = solidShapes[lnkIndex].size();
+
+        // If there are not shapes, we just continue
+        if (nrOfSolidShapes==0)
+        {
+            continue;
+        }
+
+        if (nrOfSolidShapes == 1)
+        {
+            // In this case, if there is no valid name the shape will just be called <linkName>_collision or <linkName>_visual
```

Comment:
Is there a particular reason why you use suffixes instead of prefixes? 
I found prefixes easier to filter, but maybe it is a personal preference

Reply:
I actually wanted to ask this to you. No reason, and I would prefer to avoid configurability to avoid getting crazy. If @CarlottaSartore agrees on prefixes, I think we can switch to prefixes.

Reply:
Actually I think we are already using suffixes in sensor names (like `l_arm_imu_0`), so probably we may want to be consistent with that. I may end with an option in the end, even if I wanted to avoid it. :D

Reply:
Apart from icub models convention, from my side it is fine, if you decided with prefix I will change accordingly https://github.com/ami-iit/comodo/pull/9/files#diff-8c6f7f502ad9d4b4144a518db978a03162715053482739674d0703a28fb9ef85R225-R235 to be coherent

Reply:
> Actually I think we are already using suffixes in sensor names (like `l_arm_imu_0`) in xCub models, so probably we may want to be consistent with that. I may end with an option in the end, even if I wanted to avoid it. :D

Yes, I think it is better to use suffixes at least in iCub models. @francesco-romano I can easly add support also for prefix, just to understand for links with multiples shapes would you prefer: `collision_linkName_1` or `collision_1_linkName` ?

Reply:
No worries. Go with suffixes. I am manually assign all names so I am not planning to use this function at the moment.
I was only suggesting if there was a specific reason and it seems there is :)

Reply:
Ok, great! In the future we can add variants to this naming scheme as necessary.